state_server.go

```diff
@@ -0,0 +1,185 @@
+package terminal
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/improbable-eng/grpc-web/go/grpcweb"
+	"github.com/lightninglabs/lightning-terminal/litrpc"
+	"google.golang.org/grpc"
+)
+
+const (
+	LitSubServer     string = "lit"
+	LNDSubServer     string = "lnd"
+	PoolSubServer    string = "pool"
+	LoopSubServer    string = "loop"
+	FaradaySubServer string = "faraday"
+)
+
+// statusServer is an implementation of the litrpc.StatusServer which can be
+// queried for the status of various LiT sub-servers.
+type statusServer struct {
+	litrpc.UnimplementedStatusServer
+
+	m  map[string]*subServerStatus
+	mu sync.RWMutex
+
+	grpcServer   *grpc.Server
+	grpcWebProxy *grpcweb.WrappedGrpcServer
+}
+
+// subServerStatus represents the status of a sub-server.
+type subServerStatus struct {
+	running bool
+	err     string
+}
+
+// newSubServerStatus constructs a new subServerStatus.
+func newSubServerStatus() *subServerStatus {
+	return &subServerStatus{}
+}
+
+// newStatusServer constructs a new statusServer.
+func newStatusServer() *statusServer {
+	s := &statusServer{
+		m: map[string]*subServerStatus{
+			LitSubServer:     newSubServerStatus(),
+			LNDSubServer:     newSubServerStatus(),
+			FaradaySubServer: newSubServerStatus(),
+			PoolSubServer:    newSubServerStatus(),
+			LoopSubServer:    newSubServerStatus(),
+		},
+	}
+
+	options := []grpcweb.Option{
```

Comment:
Do we really need to spin up our own grpc-web and gRPC server? Can't we instead have this as a sub component of the `rpcProxy` that already does all of this for us?

Reply:
Just realised why I did it this way initially - the reason is that if we fail to start LND, then we never get to the registration and serving of the status server. So I think we need to re-add this actually